flink-architecture-tests/flink-architecture-tests-test/src/main/java/org/apache/flink/architecture/rules/ITCaseRules.java

```diff
@@ -130,7 +130,7 @@
 
     private static DescribedPredicate<JavaClass> miniClusterExtensionRule() {
         return containAnyFieldsInClassHierarchyThat(
-                arePublicStaticFinalAssignableTo(MiniClusterExtension.class));
+                arePublicStaticFinalAssignableTo(InternalMiniClusterExtension.class));
```

Comment:
This will need further adjustments to allow private fields, and it should check that we actually use RegisterExtension.
Also needs to cover ExtendWith(..)